pyro/infer/mcmc/api.py

```diff
@@ -231,8 +234,9 @@ def run(self, *args, **kwargs):
                     yield val, chain_id
                 else:
                     active_workers -= 1
+            exc_raised = False
         finally:
-            self.terminate()
+            self.terminate(terminate_workers=exc_raised)
 
 
 class MCMC(object):
```

Comment:
could you update the docs for `hook_fn`?

Reply:
Oops..thanks for catching this.